test/CurveFactoryV2.t.sol

```diff
@@ -172,4 +168,94 @@ contract CurveFactoryV2Test is Test {
         cheats.prank(address(liquidityProvider));
         dfxCadcCurve.withdraw(100_000e18, block.timestamp + 60);
     }
+
+    function test_depositGlobalGuard() public {
```

Comment:
100k and 110k are good to test each function but just as a sanity check can you run this through some Fuzzing?